jina/executors/encoders/audio/spectral.py

```diff
@@ -0,0 +1,89 @@
+        :param n_mfcc: the number of coefficients (20 by default)
+        :param n_fft: length of the FFT window (2048 by default)
+        :param hop_length: the number of samples between successive MFCC frames (512 by default)
+        """
+        super().__init__(*args, **kwargs)
+        self.input_sample_rate = input_sample_rate
+        self.n_mfcc = n_mfcc
+        self.n_fft_length = n_fft_length
+        self.hop_length = hop_length
+
+    @batching
+    @as_ndarray
+    def encode(self, data: np.ndarray, *args, **kwargs) -> np.ndarray:
+        """
+        Segments the audio signal of each Chunk into short MFCC frames, extracts MFCCs for each frame and concatenates
+        Chunk frame MFCCs into a single Chunk embedding.
+
+        :param data: a `Batch x Signal Length` ndarray, where `Signal Length` is a number of samples
+        :return: a `Batch x Concatenated Features` ndarray, where `Concatinated Features` is a `n_mfcc`-dimensional
+        feature vector times the number of the MFCC frames
+        """
+        import librosa
+        embeds = []
+        for chunk_data in data:
+            mfccs = librosa.feature.mfcc(y=chunk_data, sr=self.input_sample_rate, n_mfcc=self.n_mfcc,
+                                         n_fft=self.n_fft_length, hop_length=self.hop_length)
+            embeds.append(mfccs.flatten())
+        result = np.array(embeds)
+        return result
+
+
+class ChromaPitchEncoder(BaseAudioEncoder):
+    """
+    :class:`ChromaPitchEncoder` is based on chroma spectrograms (chromagrams) which represent melodic/harmonic features.
+    :class:`ChromaPitchEncoder` encodes an audio signal from a `Batch x Signal Length` ndarray into a
+    `Batch x Concatenated Features` ndarray.
+    """
+
+    def __init__(self, input_sample_rate: int = 22050, hop_length: int = 512, *args, **kwargs):
+        """
+        :class:`ChromaPitchEncoder` extracts from an audio signal 12-dimensional feature vectors, that refer to 12
+        octaves, and concatenate the vectors.
+
+        :param input_sample_rate: input sampling rate in Hz (22050 by default)
+        :param hop_length: the number of samples between successive chroma frames (512 by default)
+        """
+        super().__init__(*args, **kwargs)
+        self.input_sample_rate = input_sample_rate
+        self.hop_length = hop_length
+
+    @batching
+    @as_ndarray
+    def encode(self, data: np.ndarray, *args, **kwargs) -> np.ndarray:
+        """
+        Segments the audio signal of each Chunk into short chroma frames, extracts chromagrams for each frame and
+        concatenates Chunk frame chromagrams into a single Chunk embedding.
+
+        :param data: a `Batch x Signal Length` ndarray, where `Signal Length` is a number of samples
+        :return: a `Batch x Concatenated Features` ndarray, where `Concatinated Features` is a 12-dimensional feature
+        vector times the number of the chroma frames
+        """
+        import librosa
```

Comment:
```suggestion
        from librosa.feature import chroma_cqt
```
The same change goes here. We can wrap this function in `post_init` as well so that we won't repeat importing when calling encoding.